### PR TITLE
refact(modules): add support for v2 major release go-modules

### PR DESCRIPTION
**Pull request by prateekpandey14**
**Why we need this:**
- If we  are releasing a v2 or higher module, we need have follow the [steps](https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher) to correctly represents the semantic version for v2 or higher release modules. otherwise its just shows the `v1.12.xxxx` as latest releases at the user end not `v2.0.xxx` 

- Another way to handle this not releasing the v2.x.x modules, let the release in v1.x.x semantic version then we don't have to do this changes. 

**What this PR does:**

Changes follows the one of the way to release a major modules , more info https://github.com/golang/go/wiki/Modules#releasing-modules-v2-or-higher

- Update the go.mod file to include a /v2 at the end of the module path in the module directive (e.g., module github.com/openebs/api/v2). 
- Update import statements within the module to also use /v2 (ex., import "github.com/openebs/api/v2/mypkg"). 
- Update Makefile to handle the code-gen and conversion funcs.

**How to Release New Major modules:**

Let us say we want to release v3 then following are the steps: 

- Replace the `v2` to `v3` in go.mod module path.
- Update import path as `github.com/openebs/api/v3`
- Create a new Tag for major release from master branch ex: `v3.0.0`

Refer for more details: https://github.com/openebs/cstor-operators/issues/164

Signed-off-by: prateekpandey14 <prateek.pandey@mayadata.io>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,6 +39,11 @@ generate-crds: controller-gen
 # code generation for custom resources
 kubegen:
 	./hack/update-codegen.sh
+	@cp -rf v2/pkg/apis pkg/
+	@cp -rf v2/pkg/client pkg/
+	@cp -rf v2/pkg/internalapis/apis/cstor/v1/zz_generated.conversion.go pkg/internalapis/apis/cstor/v1/zz_generated.conversion.go
+	@cp -rf v2/pkg/internalapis/apis/cstor/zz_generated.deepcopy.go pkg/internalapis/apis/cstor/zz_generated.deepcopy.go
+	@rm -rf v2
```
Comment (akhilerm) [inline Makefile:46]: question: Is this because the codegen generates the v2 directory ??

File: go.mod [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module github.com/openebs/api
+module github.com/openebs/api/v2
 
 go 1.13
```
Comment (akhilerm) [inline go.mod]: go version need to be 1.14.7 or above, because of [CVE-2020-16845](https://nvd.nist.gov/vuln/detail/CVE-2020-16845)

Comment (prateekpandey14) [reply go.mod]: that's a go.mod file automatically generated at go 1.13.x version timeframe using `go mod init`

File: Makefile (hunk 1 above)
Comment (prateekpandey14) [reply Makefile:46]: Yes, all the generated `clientset, informers, deepcopy` paths has to contain the module path `github.com/openebs/api/v2/...`  so we have to use the same module name in code-gen as well, which creates the `v2` directory and then i moved all updated required files to the exact location and remove it.

File: go.mod (hunk 2 above)
Comment (prateekpandey14) [reply go.mod]: But we can update the version to `go 1.14` as well .

**Review by akhilerm — COMMENTED** [review]
given some comments.

**Review by akhilerm — APPROVED** [review]
/lgtm

**Review by sonasingh46 — APPROVED** [review]

**Pull request merged**